clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/agents/TitusClusterCachingAgent.groovy

```diff
@@ -83,6 +98,87 @@ class TitanClusterCachingAgent implements CachingAgent { //, OnDemandAgent {
     types
   }
 
+  @Override
+  String getOnDemandAgentType() {
+    "${getAgentType()}-OnDemand"
+  }
+
+  @Override
+  OnDemandMetricsSupport getMetricsSupport() {
+    return metricsSupport
+  }
+
+  @Override
+  boolean handles(String type) {
+    return false
+  }
+
+  @Override
+  boolean handles(String type, String cloudProvider) {
+    return type == ON_DEMAND_TYPE && cloudProvider == titusCloudProvider.id
+  }
+
+  @Override
+  OnDemandAgent.OnDemandResult handle(ProviderCache providerCache, Map<String, ? extends Object> data) {
+    if (!data.containsKey("serverGroupName")) {
+      return null
+    }
+    if (!data.containsKey("account")) {
+      return null
+    }
+    if (!data.containsKey("region")) {
+      return null
+    }
+
+    if (account.name != data.account) {
+      return null
+    }
+
+    if (region != data.region) {
+      return null
+    }
+
+    //TODO(cfieber) - this should just load a single server group and follow the same on-demand caching behaviour as the AWS provider
```

Comment:
OnDemand is non-optional now for the cluster-wide tasks since they inject the forceCacheRefreshes. This is a dirty implementation that just reloads the entire data set which works for now since the getAllJobs is filtering only service jobs of which there are currently near zero